### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by djsmallmac**
Hi

### CODE REVIEW

File: Testing.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+            System.Console.WriteLine("\n\tDeal one card\n");
+            Pack.Deal().Currentcard();
+
+            System.Console.WriteLine("\n\tShow deck\n");
+            Pack.Showcards();
+
+            System.Console.WriteLine("\n\tDealing a number of cards at a time\n");
+
+            Pack.DealCard(5);
```
Comment (djsmallmac) [inline Testing.cs:47]: Could implement a feature where the user can input how many cards they want dealt to them

File: Testing.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
+﻿using System.Runtime.ConstrainedExecution;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CMP1903MA012223
+{
+    internal class Testing
```
Comment (djsmallmac) [inline Testing.cs:10]: Testing LGTM

File: Testing.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        public static void testing()
+        {
+            
+            //Creating pack 
+            Pack deck = new Pack();
+
+            //Default arrangement of the cards
+            System.Console.WriteLine("\n\tDefault arrangement\n");
```
Comment (djsmallmac) [inline Testing.cs:19]: Good job making it so the output is on separate  lines makes it look more clean

File: Pack.cs [hunk 4]
```diff
@@ -4,30 +4,115 @@
...
+
+            else if (typeOfShuffle == 3)
+            {
 
+                System.Console.WriteLine("\nYou have chosen not to shuffle the deck of cards\n");
+                return true;
+            }
+            else
+            {
+                return false;
```
Comment (djsmallmac) [inline Pack.cs:78]: Good use of error handling to make sure that if the input is not one that is selected it will return false

File: Card.cs [hunk 5]
```diff
@@ -1,18 +1,82 @@
...
+
+            set
+            {
+                if (value >= 1 || value <= 13)
+                {
+                    val = value;
+                }
+                else
+                {
+                    System.Console.WriteLine("That is out of bounds, value must be between 1 - 13");
```
Comment (djsmallmac) [inline Card.cs:50]: Good use of error handling to make sure the value is in range

File: Pack.cs [hunk 6]
```diff
@@ -4,30 +4,115 @@
...
+                    i = 1;
+                    j++;
+                }
+            }
+        }
+        
+        public static bool ShuffleCardPack(int typeOfShuffle)
         {
             //Shuffles the pack based on the type of shuffle
+            // 1. Fisher-Yates shuffle
```
Comment (djsmallmac) [inline Pack.cs:39]: comments are adequate

File: Pack.cs [hunk 7]
```diff
@@ -4,30 +4,115 @@
 using System.Text;
 using System.Threading.Tasks;
 
-namespace CMP1903M_A01_2223
+namespace CMP1903MA012223
 {
     class Pack
```
Comment (djsmallmac) [inline Pack.cs:9]: LGTM

File: Card.cs [hunk 8]
```diff
@@ -1,18 +1,82 @@
...
 using System.Text;
 using System.Threading.Tasks;
 
-namespace CMP1903M_A01_2223
+
+
+
+namespace CMP1903MA012223
 {
     class Card
```
Comment (djsmallmac) [inline Card.cs:15]: LGTM

**Review by djsmallmac — COMMENTED** [review]
1. How is the code documented:
A good number of commits allows to see how the code progressed through different stages.
Comments could be more descriptive of what each parts does.
Naming conventions are easy to follow.
2. How are errors handled:
There is multiple instances where errors are handled through validation of inputs, this is done very well as I cannot see any other puts where an error can slip through.
3. How can the code be improved:
You could add more features such as hands and more shuffles.
Comments could be more fleshed out telling what each part does.
You could make it so the user is able to input how many cards they want dealt or what kind of shuffle they want.

**Comment by colecrawley** [comment]
Glad you enjoyed! Tune in for next week where we take a look at Elle Julia's code in the next episode of "clap" Code "clap" Review! Thanks for watching, subscribe for more!

**Pull request closed without merging**